### PR TITLE
Troubleshooting

### PR DESCRIPTION
**Pull request by brey**
BOUNDARY: Fix bug  & manage workflow to better handle coasts=None.
DEM: Include option to pass an xarray Dataset as argument. Fix management of global setup. Update tests to include more cases.
TESTS: move errdap tests to --runslow

### CODE REVIEW

File: pyposeidon/boundary.py [hunk 1]
```diff
@@ -22,11 +22,11 @@ def simplify(geo):
     geo = geo.loc[:, ["geometry"]].explode(index_parts=True).droplevel(0).reset_index(drop=True)
 
     # Just in case merge
```
Comment (pmav99) [inline pyposeidon/boundary.py:24]: If this is something that normally shouldn't happen, perhaps adding a log message would not be a bad idea. Might help identify some issue

File: tests/test_dem.py [hunk 2]
```diff
@@ -1,47 +1,57 @@
...
+        pytest.param({"lon_min": -360.0, "lon_max": 0, "lat_min": -90.0, "lat_max": 90.0}, (600,1200), id="global 1"),
         # fmt: on
     ],
 )
 
 
 @DEM_SOURCES
 @WINDOWS
-def test_dem(dem_source, kwargs):
+def test_dem(dem_source, kwargs, expected):
```
Comment (pmav99) [inline tests/test_dem.py:40]: Shouldn't we use `expected` in this function?

Comment (brey) [reply tests/test_dem.py:40]: The one function tests the shape of the output and the other the existence of NaNs.

File: pyposeidon/boundary.py (hunk 1 above)
Comment (brey) [reply pyposeidon/boundary.py:24]: This relates to the quality of the input coastlines. This is a function to  check/fix any inconsistencies but it is not bulletproof. Generally speaking, it is the user who should provide a quality coastlines dataset.

**Pull request merged**